fs_mark/fs_mark.py

```diff
@@ -0,0 +1,197 @@
+import trunner
+import time
+import os
+import yaml
+import re
+import pexpect
+import timing_data as t_data
+import psh.tools.psh as psh
+
+from datetime import datetime
+from trunner.ctx import TestContext
+from trunner.dut import Dut
+from trunner.types import TestResult, TestSubResult, Status
+
+
+def clean(dut, test_name, ctx: TestContext) -> int:
+    # Dummyfs doesn't need clean since target is rebooted after every test
+    if not ctx.target.rootfs:
+        return 0
+
+    if '-k' in test_name or '-L' in test_name or '-F' in test_name:
+        args = test_name.split()
+        args = args[1:]
+
+        def get_dir(x):
+            if args[x[0] - 1] == '-d':
+                return x[1]
+
+        dirs = list(map(lambda x: get_dir(x), enumerate(args)))
+        dirs = [x for x in dirs if x is not None]
+
+        dut.expect(re.escape(ctx.target.shell_prompt), timeout=120)
+        psh._send(dut, f'/bin/fs_mark_clean {" ".join(dirs)}')
+        try:
+            idx = dut.expect([re.escape(ctx.target.shell_prompt),
+                              r".+?Error in remove_dir_recursive.+?"], timeout=1800)
+
+            if idx == 1:
+                return -1
+            elif idx == 0:
+                return 0
+        except pexpect.TIMEOUT:
+            return -2
+    else:
+        return 0
+
+
+def harness(dut: Dut, ctx: TestContext, result: TestResult):
+    target = trunner.ctx.target.name
+    logs_path = '~/remote/logs/fs_mark_logs_test'
+    test_status = Status.OK
+
+    subresult = None
+    test_name = None
+    got_time = False
+    first_loop = True
+
+    f_use = None
+    count = None
+    size = None
+    files_sec = None
+    app_overhead = None
+
+    start_time = None
+    end_time = None
+    loop_time = 1200
+
+    NAME = r".+?#  (?P<name>(/bin/)?fs_mark.+?)\r+\n"
+    MSG_LINE = r"(?P<line>(\s+\d+){3}.+?)\r+\n"
+    NO_SPC = r"Insufficient free space.+?\r+\n"
+    END = r"Average Files/sec:.+?\r+\n"
+    ERR = r".+?(?P<err>EIO|ENOSPC|ENOMEM).+?"
+    NO_CONT_BLOCK = r"(?P<msg>Lack of contiguous memory block of size \d+ bytes.+?)\r+\n"
+
+    while True:
+        if start_time and end_time:
+            loop_time = 3 * (end_time - start_time)
+            start_time = None
+            end_time = None
+
+        try:
+            idx = dut.expect([
+                NAME,
+                MSG_LINE,
+                NO_SPC,
+                END,
+                ERR,
+                NO_CONT_BLOCK
+            ], timeout=loop_time)
+            parsed = dut.match.groupdict()
+
+        except pexpect.TIMEOUT:
+            test_status = Status.FAIL
+            subresult = TestSubResult(name=str(test_name), msg='Timeout, probably fs hang-up', status=Status.FAIL)
+            result.add_subresult_obj(subresult)
```

Comment:
There's no longer `add_subresult_obj`, use `add_subresult` instead, so also no need to use TestSubResult